ibl_to_nwb/brainwide_map/read_after_write.py

```diff
@@ -0,0 +1,361 @@
+import os
+import numpy as np
+import pandas as pd
+from pathlib import Path
+import h5py
+from pynwb import NWBHDF5IO, NWBFile
+from one.api import ONE
+import logging
+from typing import Optional
+from numpy import testing
+
+def check_arrays(array_a: np.ndarray, array_b: np.ndarray, full_check: bool = False):
+    """checks if two arrays contain the same numerical values
+
+    Args:
+        array_a (np.ndarray): _description_
+        array_b (np.ndarray): _description_
+        full_check (bool, optional): If True, compares all values of the arrays. If False, checks a small subsample only. Defaults to False.
+    """
+
+    # if full check, check all samples
+    if full_check:
+        testing.assert_allclose(array_a, array_b)
+
+    # check just a random subset of samples
+    else:
+        inds = np.random.randint(0, np.prod(array_a.shape), size=10)
+        testing.assert_allclose(np.ravel(array_a)[inds], np.ravel(array_b)[inds])
+        
+
+
+def check_series(series_a: pd.Series, series_b: pd.Series, full_check: bool = False):
+    """checks if two pd.Series contain the same numerical values. Checks if NaN values are the same.
+
+    Args:
+        series_a (pd.Series): _description_
+        series_b (pd.Series): _description_
+        full_check (bool, optional): _description_. Defaults to False.
+    """
+
+    # -> all of this functionality is now moved to check_arrays()
+    # this function as of now is obsolete but kept for potential future integration
+    
+    check_arrays(series_a.values, series_b.values, full_check=full_check)
+
+
+def check_tables(table_a: pd.DataFrame, table_b: pd.DataFrame, naming_map: dict = None, full_check: bool = False):
```

Comment:
If this is for checking that two pandas data frames are equivalent, why not use`pandas.testing.assert_frame_equal`? (and resolve any column name mapping by creating a copy of the NWB or ONE frame and renaming the columns on the copy passed to the assertion?)

Reply:
:) thanks for pointing me towards `pandas.testing`